types/index.d.ts

```diff
@@ -690,9 +694,10 @@ export interface QueryCache {
   }): Promise<TResult>
 
   getQueryData<T = unknown>(key: AnyQueryKey | string): T | undefined
-  setQueryData<T = unknown>(
```

Comment:
Not sure if changing the `<T = unknown>` to `<TResult, TError>` could be problematic here.